fusedrug/utils/file_formats/fasta_efficient_db_index.py

```diff
@@ -0,0 +1,174 @@
+from typing import Optional, Callable
+import mmap
+import os
+import sqlite3
+import threading
+import multiprocessing
+
+g_lock = multiprocessing.Lock()
+
+
+class FastFASTAReader:
+    def __init__(
+        self,
+        fasta_path: str,
+        index_file_path: Optional[str] = None,
+        entry_id_process_func: Optional[Callable[[str], str]] = None,
+    ):
+        """
+        Create an optimized index for rapid FASTA sequence lookup.
+
+        Args:
+            fasta_path (str): Path to FASTA file
+            index_file_path (str): Optional. The file path of the index file. If not provided, will defauylt to {fasta_path}.sqlite
```

Comment:
Add details about entry_id_process_func. What is it for? what are the expected arguments and returned value?

Reply:
done